### PR TITLE
DOCS-4054 React Native Monitoring Advanced Configuration Documentation

### PR DESCRIPTION
**Pull request by alai97**
### What does this PR do?

Adds single sourcing logic and generates a preview for an Advanced Configuration doc page.

### Motivation

DOCS-4054

### Additional Notes

Remember to update the `pull_config.yaml` back to `main` before merging!

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: local/bin/py/build/configurations/pull_config.yaml [hunk 1]
```diff
@@ -735,13 +735,31 @@
...
+              text: "Learn how to explore your RUM data"
+    - action: pull-and-push-file
+      branch: alai97/reactnative-advanced-configuration-doc-update
+      globs:
+      - 'docs/advanced_configuration.md'
+      options:
+        dest_path: '/real_user_monitoring/reactnative/'
+        file_name: 'advanced_configuration.md'
+        front_matters:
+          title: React Native Advanced Configuration
```
Comment (alai97) [inline local/bin/py/build/configurations/pull_config.yaml:752]:
```suggestion
          title: RUM React Native Advanced Configuration
```

File: local/bin/py/build/configurations/pull_config_preview.yaml [hunk 2]
```diff
@@ -738,12 +738,31 @@
...
+              text: "Learn how to explore your RUM data"
+    - action: pull-and-push-file
+      branch: alai97/reactnative-advanced-configuration-doc-update
+      globs:
+      - 'docs/advanced_configuration.md'
+      options:
+        dest_path: '/real_user_monitoring/reactnative/'
+        file_name: 'advanced_configuration.md'
+        front_matters:
+          title: React Native Advanced Configuration
```
Comment (alai97) [inline local/bin/py/build/configurations/pull_config_preview.yaml:755]:
```suggestion
          title: RUM React Native Advanced Configuration
```

**Review by alai97 — COMMENTED** [review]
For consistency of Advanced Configuration page titles.

**Review by cswatt — APPROVED** [review]

File: local/bin/py/build/configurations/pull_config.yaml [hunk 3]
```diff
@@ -735,13 +735,31 @@
...
+      globs:
+      - 'docs/advanced_configuration.md'
+      options:
+        dest_path: '/real_user_monitoring/reactnative/'
+        file_name: 'advanced_configuration.md'
+        front_matters:
+          title: RUM React Native Advanced Configuration
+          kind: documentation
+          description: Learn about advanced configuration options for your React Native setup.
+          dependencies: ["https://github.com/DataDog/dd-sdk-reactnative/blob/alai97/reactnative-advanced-configuration-doc-update/docs/advanced_configuration.md"]
```
Comment (alai97) [inline local/bin/py/build/configurations/pull_config.yaml:755]:
```suggestion
          dependencies: ["https://github.com/DataDog/dd-sdk-reactnative/blob/main/docs/advanced_configuration.md"]
```

File: local/bin/py/build/configurations/pull_config.yaml [hunk 4]
```diff
@@ -735,13 +735,31 @@
...
-              text: "Learn how to explore your RUM data"
             - link: "https://www.datadoghq.com/blog/react-native-monitoring/"
               tag: "Blog"
               text: "Monitor React Native applications"
-
+            - link: "real_user_monitoring/explorer/"
+              tag: "Documentation"
+              text: "Learn how to explore your RUM data"
+    - action: pull-and-push-file
+      branch: alai97/reactnative-advanced-configuration-doc-update
```
Comment (alai97) [inline local/bin/py/build/configurations/pull_config.yaml:745]:
```suggestion
      branch: main
```

File: local/bin/py/build/configurations/pull_config_preview.yaml [hunk 5]
```diff
@@ -738,12 +738,31 @@
...
-              tag: "Documentation"
-              text: "Learn how to explore your RUM data"
             - link: "https://www.datadoghq.com/blog/react-native-monitoring/"
               tag: "Blog"
               text: "Monitor React Native applications"
+            - link: "real_user_monitoring/explorer/"
+              tag: "Documentation"
+              text: "Learn how to explore your RUM data"
+    - action: pull-and-push-file
+      branch: alai97/reactnative-advanced-configuration-doc-update
```
Comment (alai97) [inline local/bin/py/build/configurations/pull_config_preview.yaml:748]:
```suggestion
      branch: main
```

File: local/bin/py/build/configurations/pull_config_preview.yaml [hunk 6]
```diff
@@ -738,12 +738,31 @@
...
+      globs:
+      - 'docs/advanced_configuration.md'
+      options:
+        dest_path: '/real_user_monitoring/reactnative/'
+        file_name: 'advanced_configuration.md'
+        front_matters:
+          title: RUM React Native Advanced Configuration
+          kind: documentation
+          description: Learn about advanced configuration options for your React Native setup.
+          dependencies: ["https://github.com/DataDog/dd-sdk-reactnative/blob/alai97/reactnative-advanced-configuration-doc-update/docs/advanced_configuration.md"]
```
Comment (alai97) [inline local/bin/py/build/configurations/pull_config_preview.yaml:758]:
```suggestion
          dependencies: ["https://github.com/DataDog/dd-sdk-reactnative/blob/main/docs/advanced_configuration.md"]
```

**Pull request merged**